### PR TITLE
Revert deconstruction for SMTP commands

### LINKED ISSUE #2398: PR #2250 Issues
#2250 is causing some weird issues that I'm having trouble tracking down.

`````
3/27/2018 2:44:39 PM[CRIT] [C5B3571A-110A-4942-8A47-DEB528847811] [core] Plugin auth_trap failed: TypeError: /^([^ ]*)( +(.*))?$/.exec is not a function or its return value is not iterable
3/27/2018 2:44:39 PM    at Connection.process_line (/usr/local/lib/node_modules/Haraka/connection.js:315:77)
3/27/2018 2:44:39 PM    at Connection._process_data (/usr/local/lib/node_modules/Haraka/connection.js:464:22)
3/27/2018 2:44:39 PM    at Connection.respond (/usr/local/lib/node_modules/Haraka/connection.js:561:14)
3/27/2018 2:44:39 PM    at Connection.unrecognized_command_respond (/usr/local/lib/node_modules/Haraka/connection.js:750:22)
3/27/2018 2:44:39 PM    at callback (/usr/local/lib/node_modules/Haraka/plugins.js:480:35)
3/27/2018 2:44:39 PM    at Plugin.exports.hook_unrecognized_command (/usr/local/lib/node_modules/Haraka/plugins/auth/auth_base.js:35:52)
3/27/2018 2:44:39 PM    at Object.plugins.run_next_hook (/usr/local/lib/node_modules/Haraka/plugins.js:505:28)
3/27/2018 2:44:39 PM    at callback (/usr/local/lib/node_modules/Haraka/plugins.js:463:32)
3/27/2018 2:44:39 PM    at Plugin.exports.upgrade_connection (/usr/local/lib/node_modules/Haraka/plugins/tls.js:90:44)
3/27/2018 2:44:39 PM    at Object.plugins.run_next_hook (/usr/local/lib/node_modules/Haraka/plugins.js:505:28)
`````

and

`````
4/18/2018 6:31:25 PM[CRIT] [E28E7A65-B41C-4A69-B835-6E5716E10C30] [core] Plugin auth_trap failed: TypeError: Cannot read property 'Symbol(Symbol.iterator)' of null
4/18/2018 6:31:25 PM    at Connection.process_line (/usr/local/lib/node_modules/Haraka/connection.js:315:77)
4/18/2018 6:31:25 PM    at Connection._process_data (/usr/local/lib/node_modules/Haraka/connection.js:464:22)
4/18/2018 6:31:25 PM    at Connection.respond (/usr/local/lib/node_modules/Haraka/connection.js:561:14)
4/18/2018 6:31:25 PM    at Connection.unrecognized_command_respond (/usr/local/lib/node_modules/Haraka/connection.js:750:22)
4/18/2018 6:31:25 PM    at callback (/usr/local/lib/node_modules/Haraka/plugins.js:480:35)
4/18/2018 6:31:25 PM    at Plugin.exports.hook_unrecognized_command (/usr/local/lib/node_modules/Haraka/plugins/auth/auth_base.js:35:52)
4/18/2018 6:31:25 PM    at Object.plugins.run_next_hook (/usr/local/lib/node_modules/Haraka/plugins.js:505:28)
4/18/2018 6:31:25 PM    at callback (/usr/local/lib/node_modules/Haraka/plugins.js:463:32)
4/18/2018 6:31:25 PM    at Plugin.exports.upgrade_connection (/usr/local/lib/node_modules/Haraka/plugins/tls.js:90:44)
4/18/2018 6:31:25 PM    at Object.plugins.run_next_hook (/usr/local/lib/node_modules/Haraka/plugins.js:505:28)
4/18/2018 6:31:25 PM    at callback (/usr/local/lib/node_modules/Haraka/plugins.js:463:32)
4/18/2018 6:31:25 PM    at Plugin.exports.hook_unrecognized_command (/etc/haraka/plugins/xclient.js:43:16)
4/18/2018 6:31:25 PM    at Object.plugins.run_next_hook (/usr/local/lib/node_modules/Haraka/plugins.js:505:28)
4/18/2018 6:31:25 PM    at Object.plugins.run_hooks (/usr/local/lib/node_modules/Haraka/plugins.js:422:13)
4/18/2018 6:31:25 PM    at Connection.process_line (/usr/local/lib/node_modules/Haraka/connection.js:342:25)
4/18/2018 6:31:25 PM    at Connection._process_data (/usr/local/lib/node_modules/Haraka/connection.js:464:22)
4/18/2018 6:31:25 PM    at Connection.respond (/usr/local/lib/node_modules/Haraka/connection.js:561:14)
4/18/2018 6:31:25 PM    at Connection.connect_respond (/usr/local/lib/node_modules/Haraka/connection.js:793:22)
4/18/2018 6:31:25 PM    at callback (/usr/local/lib/node_modules/Haraka/plugins.js:480:35)
4/18/2018 6:31:25 PM    at Plugin.exports.hook_connect (/etc/haraka/plugins/abusix_stats.js:19:12)
4/18/2018 6:31:25 PM    at Object.plugins.run_next_hook (/usr/local/lib/node_modules/Haraka/plugins.js:505:28)
4/18/2018 6:31:25 PM    at Object.plugins.run_hooks (/usr/local/lib/node_modules/Haraka/plugins.js:422:13)
4/18/2018 6:31:25 PM    at Connection.rdns_response (/usr/local/lib/node_modules/Haraka/connection.js:729:17)
4/18/2018 6:31:25 PM    at QueryReqWrap.dns.reverse [as callback] (/usr/local/lib/node_modules/Haraka/connection.js:708:26)
`````

My vote would be to just revert it as the old code did not have these issues.

### PR DESCRIPTION
**Pull request by smfreegard**
Fixes #2398 

Checklist:
- [X] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

File: connection.js [hunk 1]
```diff
@@ -351,7 +351,9 @@ class Connection {
             }
             else {
                 // unrecognized command
-                plugins.run_hooks('unrecognized_command', this, [methodMatch, remainingMatch]);
+                matches.splice(0,1);
+                matches.splice(1,1);
+                plugins.run_hooks('unrecognized_command', this, matches);
```
Comment (baudehlo) [inline connection.js]: I don't like the splice here - seems potentially fragile. We should just use `[matches[1], matches[2]]`.

Comment (msimerson) [reply connection.js]: There's already named variables right above telling what matches[1] and matches[3] are (not matches[2], using splice isn't easy to intuit), so better would be using [method, remaining] as it was before.

Comment (baudehlo) [reply connection.js]: The variables are different since `method` is actually `cmd_${methodMatch.toLowerCase()}`.

**Comment by smfreegard** [comment]
I've done a bit of each of your suggestions.

**Pull request merged**